### PR TITLE
Post-process API Reference Docs

### PR DESCRIPTION
**Pull request by SgtCoDFish**
This adds an additional nodejs script which runs after `gen-crd-api-reference-docs`. This prettifies and formats the HTML, making it _much_ easier to read and reason about. It also includes a minor cleanup of the generation script for API docs.

This also pre-emptively fixes some issues with MDX parsing in the refreshed website and makes everything a lot easier there.

**IMPORTANT**: Note that API + CLI reference docs are broken in netlify previews and always have been! It's not possible to check [/docs](https://deploy-preview-870--cert-manager-website.netlify.app/docs/) for this, but it is possible to check older checked-in versions of the API docs, e.g. for the [1.5 release](https://deploy-preview-870--cert-manager-website.netlify.app/v1.5-docs/reference/api-docs/)

### CODE REVIEW

File: scripts/gendocs/generate [hunk 1]
```diff
@@ -163,32 +181,36 @@ genversionwithcli "release-1.7" "v1.7-docs"
...
+    rm -r pkg/apis/acme/v1alpha3
+    rm -r pkg/apis/acme/v1beta1
+    rm -r pkg/apis/certmanager/v1alpha2
+    rm -r pkg/apis/certmanager/v1alpha3
+    rm -r pkg/apis/certmanager/v1beta1
+
+    gendocs "v1.6-docs"
+}
 
+#generate_release16
```
Comment (SgtCoDFish) [inline scripts/gendocs/generate:213]: note: this simply exists to make it easier to comment and uncomment the generation of release-1.6

File: scripts/gendocs/postprocess/api-doc-postprocess.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  parser: 'babel',
+  htmlWhitespaceSensitivity: 'strict',
+  proseWrap: 'never',
+  quoteProps: 'consistent',
+  bracketSameLine: true,
+  printWidth: 1000
+})
+
+result = result.split('\n').slice(1, -2).join('\n')
+result = result.replace(new RegExp(/{" "}/g), ' ')
```
Comment (SgtCoDFish) [inline scripts/gendocs/postprocess/api-doc-postprocess.js:21]: note: these bits are the magic for the refreshed website, written by Patrick. It makes sure the docs are parse-able by the MDX stuff.

File: scripts/gendocs/postprocess/api-doc-postprocess.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env node
```
Comment (SgtCoDFish) [inline scripts/gendocs/postprocess/api-doc-postprocess.js:1]: note: this script was mostly written by [@PatrickHeneise](https://github.com/PatrickHeneise) - I changed it to use stdin and stdout, but he did the rest!

**Review by SgtCoDFish — COMMENTED** [review]
Left a few notes to make it easier to review!

File: scripts/gendocs/postprocess/api-doc-postprocess.js (hunk 2 above)
Comment (PatrickHeneise) [reply scripts/gendocs/postprocess/api-doc-postprocess.js:21]: Correct. Without explicit code blocks, the parser throws an exception, as it awaits an expression with `{}`:

```
Could not parse expression with acorn: Unexpected content after expression

  136 | > 
  137 | >     $ kubectl create secret generic eab-secret --from-literal \
> 138 | >       secret={base64 encoded secret key}
      |                       ^
  139 |
  140 | An example of an ACME issuer with an External Account Binding is as follows.
```

By setting explicit code blocks/fences (```), everything works fine.

File: scripts/gendocs/postprocess/api-doc-postprocess.js (hunk 3 above)
Comment (PatrickHeneise) [reply scripts/gendocs/postprocess/api-doc-postprocess.js:1]: Based on https://github.com/zentered/docs-manifest-generator

**Review by jakexks — APPROVED** [review]
/lgtm

**Comment by SgtCoDFish** [comment]
Thank you @jakexks  :heart:

**Pull request merged**